internal/sync/sync.go

```diff
@@ -200,10 +200,12 @@ func (s *Synchronizer) updateState(collectedDiff *CollectorDiff) error {
 	}
 
 	for contractAddress, memoryCells := range collectedDiff.stateDiff.StorageDiff {
-		for _, cell := range memoryCells {
-			if err := s.state.SetSlot(new(felt.Felt).SetString(contractAddress), cell.Address, cell.Value); err != nil {
-				return err
-			}
+		slots := make([]state.Slot, len(memoryCells))
+		for i, cell := range memoryCells {
+			slots[i] = state.Slot{Key: cell.Address, Value: cell.Value}
+		}
```

Comment:
It might be slightly more efficient to
1) allocate the memory and then, 
2) assign 

versus 

1) allocating the memory, 
2) zeroing it out, and then 
3) doing the assignment.

Plus we don't have to keep track of an iterator variable.

```suggestion
		slots := make([]state.Slot, 0, len(memoryCells))
		for _, cell := range memoryCells {
			slots = append(slots, state.Slot{Key: cell.Address, Value: cell.Value})
		}
```